src/client/actions/TypingStart.js

```diff
@@ -2,7 +2,14 @@
 
 const Action = require('./Action');
 const { Events } = require('../../util/Constants');
-const textBasedChannelTypes = ['dm', 'text', 'news', 'news_thread', 'public_thread', 'private_thread'];
+const textBasedChannelTypes = [
```

Comment:
A question I had when changing this is: isn't a GROUP_DM channel text-based? I didn't add it because I wasn't sure but I feel like it would make sense

Reply:
Bots can't actually interact with those (e.g. send a message to one) so it makes no sense for them to be text-based as it doesn't have any text-based methods. The only reference to them that exists in the lib is a partial for the data from invites *to* group DMs.

Reply:
Ah alright, that makes sense

Reply:
This should be a global constant at this point (in the Constants file) so you can use it in other places as mentioned.